compute_modules/function_registry/function_registry.py

```diff
@@ -22,14 +22,15 @@
 FUNCTION_SCHEMAS: List[ComputeModuleFunctionSchema] = []
 FUNCTION_SCHEMA_CONVERSIONS: Dict[str, PythonClassNode] = {}
 IS_FUNCTION_CONTEXT_TYPED: Dict[str, bool] = {}
+STREAMING: Dict[str, bool] = {}
 
 
-def add_functions(*args: Callable[..., Any]) -> None:
-    for function_ref in args:
-        add_function(function_ref=function_ref)
+def add_functions(*args: tuple[Callable[..., Any], bool]) -> None:
+    for function_ref, streaming in args:
+        add_function(function_ref=function_ref, streaming=streaming)
 
 
-def add_function(function_ref: Callable[..., Any]) -> None:
+def add_function(function_ref: Callable[..., Any], streaming: bool) -> None:
```

Comment:
Same thing for these functions - these are public apis so ideally I'd like to maintain backwards compatibility. 

`add_function` is easy - we can make `streaming` default to False.

For `add_functions` I have 2 ideas but open to others:

1. Don't allow adding streaming functions in this method. In that case someone would need to do something like this:
```python
add_function(hello_stream, streaming=True)
add_functions(
    non_stream_func_1,
    non_stream_func_2
)
```

2. Make `args` a `Union[Callable, tuple[Callable, bool]]`. Not sure if this is possible to do but if so we could iterate through args and check whether each arg is a tuple or not

Reply:
I tried union like this

``` python
def add_functions(*args: Union[Callable[..., Any], tuple[Callable[..., Any], bool]]) -> None:
    for arg in args:
        if isinstance(arg, Callable[..., Any]):
            add_function(function_ref=arg)
        elif isinstance(arg, tuple[Callable[..., Any], bool]):
            add_function(function_ref=arg[0], streaming=arg[1])
```

but it did not pass `check_mypy()` due to type issue. So I think we can go with option 1?

Reply:
yeah, works for me. I don't think it's that much of an inconvenience.